src/platform/dnne.cs

```diff
@@ -21,19 +21,55 @@
 
 namespace DNNE
 {
-    internal class ExportAttribute : System.Attribute
+    internal class ExportAttribute : Attribute
     {
         public ExportAttribute() { }
         public string EntryPoint { get; set; }
     }
 
-    internal class C99TypeAttribute : System.Attribute
+    internal class C99TypeAttribute : Attribute
     {
         public C99TypeAttribute(string code) { }
     }
 
-    internal class C99DeclCodeAttribute : System.Attribute
+    internal class C99DeclCodeAttribute : Attribute
     {
         public C99DeclCodeAttribute(string code) { }
     }
+
+    /// <summary>
+    /// Win32 'BOOL' equivalent
+    /// </summary>
+    public struct WinBool
```

Comment:
Let's stick with just strings here. The bool support is part of a larger conversation and not one that should clutter this PR. The biggest issue is DNNE would now be supplying actual .NET types that makes sharing signatures complicated.

Reply:
Once the bool logic is removed, this looks great. I would recommend looking at the work we are doing in the .NET source generator space - see https://github.com/dotnet/runtime/tree/main/docs/design/libraries/LibraryImportGenerator.